packages/shared/src/models/dataPipe/operators/concat.ts

```diff
@@ -6,24 +6,31 @@ import BaseOperator from "./base";
 type ILoadedDatalist = any; // datalist
 
 class ConcatOperator extends BaseOperator {
-  public args_list: string[];
+  public args_list: { data: any; name: string }[];
   private indexColumn = "id";
+  /** When processing arg keep track of active for logging purposes */
+  private activeArg: { data: any; name: string };
   constructor(df: DataFrame, args_list: any, pipe: DataPipe) {
     super(df, args_list, pipe);
   }
 
   // load input data list from arg, populate error object if not exist for use in validation step
   parseArg(arg: any): ILoadedDatalist {
-    return this.pipe.inputSources[arg] ?? { error: `Data list does not exists: ${arg}` };
+    const data = this.pipe.inputSources[arg];
+    if (!data) {
+      return { error: `Data list does not exists: ${arg}` };
+    }
+    return { data, name: arg };
   }
 
   validateArg(datalist: ILoadedDatalist) {
     return datalist.error ?? true;
   }
 
   apply() {
-    for (const dataList of this.args_list) {
-      this.df = this.applyConcat(dataList);
+    for (const arg of this.args_list) {
+      this.activeArg = arg;
+      this.df = this.applyConcat(arg.data);
```

Comment:
I think this is likely some confusion because of the different levels args are passed. The arg list here refers to the list of args passed to a single concat operation, i.e. the list of datalists to be concatenated as a single operation. There is a higher level set of args which refers to the full list of operations.

So in this case the concat operator has an initial dataframe (usually the output of a previous operations, or an empty dataframe if not supplied) and a list of datalists to concatenate, e.g. `list_a, list_b`.
The list is processed to retrieve the corresponding datalists (if they exist), so that each name is mapped from `name` to `{name,data}` objects. We iterate over each list and apply the concat. 

The only reason we need to track an activeArg is in case an error is thrown during the concat step (e.g. trying to concat rows with conflicting ids). Ordinarily the error would just say what id's have been found to conflict, but I thought nicer to also know whether that conflict appeared whilst processing `list_a` or `list_b`.